app/src/main/kotlin/com/simplemobiletools/clock/services/timerService.kt

```diff
@@ -0,0 +1,158 @@
+package com.simplemobiletools.clock.services
+
+import android.annotation.TargetApi
+import android.app.Notification
+import android.app.NotificationChannel
+import android.app.NotificationManager
+import android.app.Service
+import android.content.Context
+import android.content.Intent
+import android.os.Build
+import android.os.CountDownTimer
+import android.os.IBinder
+import androidx.core.app.NotificationCompat
+import com.simplemobiletools.clock.R
+import com.simplemobiletools.clock.extensions.config
+import com.simplemobiletools.clock.extensions.getOpenTimerTabIntent
+import com.simplemobiletools.clock.extensions.getTimerNotification
+import com.simplemobiletools.clock.extensions.secondsToMillis
+import com.simplemobiletools.clock.helpers.TIMER_NOTIF_ID
+import com.simplemobiletools.clock.helpers.TIMER_RUNNING_NOTIF_ID
+import com.simplemobiletools.commons.extensions.getFormattedDuration
+import com.simplemobiletools.commons.helpers.isOreoPlus
+import org.greenrobot.eventbus.EventBus
+import org.greenrobot.eventbus.Subscribe
+import org.greenrobot.eventbus.ThreadMode
+
+fun Context.startTimerService() {
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+        startForegroundService(Intent(this, TimerService::class.java))
+    } else {
+        startService(Intent(this, TimerService::class.java))
+    }
+}
+
+class TimerService : Service() {
+
+    private var timer: CountDownTimer? = null
+    private var lastTick = 0L
+    private val bus = EventBus.getDefault()
+
+    override fun onCreate() {
+        super.onCreate()
+        bus.register(this)
+    }
+
+    override fun onBind(intent: Intent?): IBinder? = null
+
+    override fun onStartCommand(intent: Intent?, flags: Int, startId: Int): Int {
+        super.onStartCommand(intent, flags, startId)
+
+        val formattedDuration = config.timerSeconds.getFormattedDuration()
+        startForeground(TIMER_RUNNING_NOTIF_ID, notification(formattedDuration))
+
+        when (val state = config.timerState) {
```

Comment:
these switch/case lines are quite confusing, simplify it please

Reply:
start from 6 fixed. Replaced toInt with roundToInt